{"version":1,"ops":[{"type":4,"author":{"id":"aef4977f5ea786238d7c9581a2f192f4adc3f39d"},"timestamp":1336220554,"metadata":{"github-id":"MDExOkNsb3NlZEV2ZW50MTQ4Mjc2MzQ="},"status":2},{"type":4,"author":{"id":"aef4977f5ea786238d7c9581a2f192f4adc3f39d"},"timestamp":1336220582,"metadata":{"github-id":"MDEzOlJlb3BlbmVkRXZlbnQxNDgyNzY0Ng=="},"status":1},{"type":3,"author":{"id":"86141a663c2eb7543d1a51ec6253a1d54de912fd"},"timestamp":1338835697,"metadata":{"github-id":"MDEyOklzc3VlQ29tbWVudDYxMDY5OTI=","github-url":"https://github.com/parrot/parrot/issues/765#issuecomment-6106992"},"message":"heap-use-after-free in interp-\u003eexit_handler_list = NULL still present, just now in Parrot_x_execute_on_exit_handlers()\n\n```\n$ ./parrot --leak-test first.pir 2\u003elog.first\n$ asan_symbolize.py \u003c log.first\n```\n\n# \n\n==30373== ERROR: AddressSanitizer heap-use-after-free on address 0x7faddba83fa0 at pc 0x7faddfa1088a bp 0x7fffff76d110 sp 0x7fffff76d108\nWRITE of size 8 at 0x7faddba83fa0 thread T0\n    #0 0x7faddfa1088a in Parrot_x_execute_on_exit_handlers /home/rurban/Perl/parrot/build-d-asan/src/exit.c:137\n    #1 0x7faddf9c8740 in Parrot_api_destroy_interpreter /home/rurban/Perl/parrot/build-d-asan/src/embed/api.c:322\n    #2 0x407086 in main /home/rurban/Perl/parrot/build-d-asan/frontend/parrot2/main.c:176\n    #3 0x7faddc3f1ead in __libc_start_main /home/aurel32/eglibc/eglibc-2.13/csu/libc-start.c:260\n0x7faddba83fa0 is located 288 bytes inside of 384-byte region [0x7faddba83e80,0x7faddba84000)\nfreed by thread T0 here:\npreviously allocated by thread T0 here:\n==30373== ABORTING\nStats: 1M malloced (1M for red zones) by 1563 calls\nStats: 0M realloced by 82 calls\nStats: 0M freed by 474 calls\nStats: 0M really freed by 0 calls\nStats: 44M (11271 full pages) mmaped in 11 calls\n  mmaps   by size class: 8:16383; 9:8191; 10:4095; 11:2047; 12:1024; 13:512; 14:256; 15:128; 16:64; 17:32; 18:16;\n  mallocs by size class: 8:1147; 9:20; 10:17; 11:204; 12:57; 13:111; 14:2; 15:1; 16:1; 17:1; 18:2;\n  frees   by size class: 8:230; 9:10; 10:7; 11:190; 12:30; 13:4; 15:1; 16:1; 17:1;\n  rfrees  by size class:\nStats: malloc large: 3 small slow: 21\nShadow byte and word:\n  0x1ff5bb7507f4: fd\n  0x1ff5bb7507f0: fd fd fd fd fd fd fd fd\nMore shadow bytes:\n  0x1ff5bb7507d0: fd fd fd fd fd fd fd fd\n  0x1ff5bb7507d8: fd fd fd fd fd fd fd fd\n  0x1ff5bb7507e0: fd fd fd fd fd fd fd fd\n  0x1ff5bb7507e8: fd fd fd fd fd fd fd fd\n=\u003e0x1ff5bb7507f0: fd fd fd fd fd fd fd fd\n  0x1ff5bb7507f8: fd fd fd fd fd fd fd fd\n  0x1ff5bb750800: fa fa fa fa fa fa fa fa\n  0x1ff5bb750808: fa fa fa fa fa fa fa fa\n  0x1ff5bb750810: 00 00 00 00 00 00 00 00\n\nI silenced the failing --leak-test with \nPARROT_NO_ADDRESS_SAFETY_ANALYSIS\non Parrot_x_execute_on_exit_handlers()","files":null},{"type":4,"author":{"id":"646f0669ead6eaf64196bebf3884a5eff162357a"},"timestamp":1338835933,"metadata":{"github-id":"MDEzOlJlb3BlbmVkRXZlbnQxNjkxMDk3NQ=="},"status":1},{"type":3,"author":{"id":"d9b4aee65ce8c18d4a43eced1eecb0d154e1eebb"},"timestamp":1339537553,"metadata":{"github-id":"MDEyOklzc3VlQ29tbWVudDYyODQwMDc=","github-url":"https://github.com/parrot/parrot/issues/765#issuecomment-6284007"},"message":"Re-opening due to new test failure in file touched by commits apparently related to this ticket:\n\n``` perl\n$ prove -v t/run/options.t \nt/run/options.t .. \n1..35\nok 1 - Start of help message\n# ...\nok 26 - -r option \u003c\"./parrot\" -D 8 -R slow \"/tmp/8VxGioq5Az.pir\" 2\u003e\u00261\u003e\nillegal argument in Parrot_interp_info\ncurrent instr.: '__show_runtime_prefix_and_exit' pc 701 (frontend/parrot2/prt0.pir:267)\nnot ok 27 - --runtime-prefix\n\n#   Failed test '--runtime-prefix'\n#   at t/run/options.t line 96.\n#                   ''\n#     doesn't match '(?^:^.+$)'\nok 28 - --gc-dynamic-threshold needs argument warning\n# ...\n# Looks like you failed 1 test of 35.\nDubious, test returned 1 (wstat 256, 0x100)\nFailed 1/35 subtests \n\nTest Summary Report\n-------------------\nt/run/options.t (Wstat: 256 Tests: 35 Failed: 1)\n  Failed test:  27\n  Non-zero exit status: 1\nFiles=1, Tests=35,  0 wallclock secs ( 0.03 usr  0.01 sys +  0.31 cusr  0.17 csys =  0.52 CPU)\nResult: FAIL\n```\n\nCan someone investigate?  Thanks.","files":null},{"type":3,"author":{"id":"86141a663c2eb7543d1a51ec6253a1d54de912fd"},"timestamp":1345796005,"metadata":{"github-id":"MDEyOklzc3VlQ29tbWVudDc5OTUwNzk=","github-url":"https://github.com/parrot/parrot/issues/765#issuecomment-7995079"},"message":"Hopefully fixed with 2162db6ed2dee7f55fa6986620e7ddfaa0a8768c\nexit via exception fooled the old logic.\nNow set and check for interp-\u003efinal_exception, similar to the already_dying logic in issue #816.","files":null},{"type":4,"author":{"id":"86141a663c2eb7543d1a51ec6253a1d54de912fd"},"timestamp":1345796006,"metadata":{"github-id":"MDExOkNsb3NlZEV2ZW50MjMyOTY5MTA="},"status":2}]}